batch_requests/concurrent/executor.py

```diff
@@ -34,40 +49,27 @@ def get_executor_pool(self, num_workers=None):
     def execute(self, requests, resp_generator,
                 result_handler=default_result_handler,
                 *args, **kwargs):
-        '''
-            Calls the resp_generator for all the requests in parallel in an asynchronous way.
-        '''
-        pool = self.get_executor_pool()
-        timeout = self.timeout
-        return [
-            result_handler(future, timeout) for future in [
-                pool.submit(resp_generator, req, *args, **kwargs)
-                for req in requests
+        '''Calls the resp_generator for all the requests in parallel in an asynchronous way'''
+        with self.get_executor_pool() as pool:
+            futures = [
+                pool.submit(resp_generator, r, *args, **kwargs)
+                for r in requests
             ]
-        ]
 
+        timeout = self.timeout
+        return [result_handler(f, timeout) for f in futures]
 
-class SequentialExecutor(Executor):
-    '''
-        Executor for executing the requests sequentially.
-    '''
 
-    def execute(self, requests, resp_generator, *args, **kwargs):
-        '''
-            Calls the resp_generator for all the requests in sequential order.
-        '''
-        return [resp_generator(request) for request in requests]
+class SequentialExecutor(Executor):
+    '''Executor for executing the requests sequentially.'''
```

Comment:
It'd be cool if this docstring matched the ones below

Reply:
Seems like a reasonable request.